src/Symfony/Component/Form/Extension/Core/Type/ChoiceType.php

```diff
@@ -302,9 +317,30 @@ public function configureOptions(OptionsResolver $resolver)
             return $choiceTranslationDomain;
         };
 
+        $multipleNormalizer = function (Options $options, $multiple) {
+            if (in_array($options['widget'], array('radio', 'checkbox'))) {
+                return 'checkbox' == $options['widget'];
+            }
+
+            return $multiple;
+        };
+
+        $expandedNomalizer = function (Options $options, $expanded) {
+            if (null !== $expanded) {
+                @trigger_error('The "expanded" option is deprecated since version 3.1 and will be removed in 4.0. You should use "widget" instead.', E_USER_DEPRECATED);
+            } else {
+                $expanded = false;
+            }
+
+            return in_array($options['widget'], array('radio', 'checkbox')) ?: $expanded;
+        };
+
         $resolver->setDefaults(array(
+            'widget' => null,
             'multiple' => false,
-            'expanded' => false,
+            'delimiter' => ',',
+            'expanded' => null, // deprecated
```

Comment:
IMHO should be marked as internal since it's still used in the implementation of this class